packages/wonder-blocks-pill/src/components/pill.tsx

```diff
@@ -0,0 +1,206 @@
+     */
+    role?: ClickableRole;
+    /**
+     * Called when the pill is clicked.
+     */
+    onClick?: () => unknown;
+    /**
+     * Custom styles to add to this pill component.
+     */
+    style?: StyleType;
+    /**
+     * Optional test ID for e2e testing.
+     */
+    testId?: string;
+};
+
+const PillInner = (props: {
+    children: string | React.ReactElement<React.ComponentProps<Typography>>;
+    size: "small" | "large";
+}) => {
+    const {children, size} = props;
+
+    if (typeof children !== "string") {
+        return children;
+    }
+
+    if (size === "small") {
+        return <LabelXSmall>{props.children}</LabelXSmall>;
+    }
+
+    return <Body>{children}</Body>;
+};
+
+/**
+ * A Pill component displays text in a rounded, colored container.
+ * This is usually used to add label tags or indicate a status.
+ *
+ * ### Usage
+ *
+ * ```jsx
+ * import Pill from "@khanacademy/wonder-blocks-pill";
+ *
+ * <Pill text="Hello, world!" />
+ * ```
+ */
+const Pill = React.forwardRef(function Pill(
+    props: Props,
+    ref: React.ForwardedRef<HTMLElement | HTMLButtonElement>,
+) {
+    const {
+        id,
+        children,
+        kind = "neutral",
+        size = "small",
+        role,
+        onClick,
+        style,
+        testId,
+    } = props;
+
+    const wrapperSizeStyle =
+        size === "small" ? pillStyles.wrapperSmall : pillStyles.wrapperLarge;
+    const colorStyles = _generateColorStyles(!!onClick, kind, size);
+
+    const defaultStyles = [
+        pillStyles.wrapper,
+        colorStyles.pill,
+        wrapperSizeStyle,
+    ];
+
+    if (onClick) {
+        return (
+            <Clickable
+                id={id}
+                role={role}
+                onClick={onClick}
+                style={[defaultStyles, colorStyles.clickableWrapper, style]}
+                testId={testId}
+                ref={ref as React.ForwardedRef<HTMLButtonElement>}
+            >
+                {() => <PillInner size={size}>{children}</PillInner>}
+            </Clickable>
+        );
+    }
+
+    return (
+        <View
+            id={id}
+            role={role}
+            style={[defaultStyles, style]}
+            testId={testId}
+            ref={ref as React.ForwardedRef<HTMLElement>}
+        >
+            <PillInner size={size}>{children}</PillInner>
+        </View>
+    );
+});
+
+const styles: Record<string, any> = {};
+
+const pillStyles = StyleSheet.create({
+    wrapper: {
+        display: "inline-flex",
+        width: "fit-content",
+    },
+    wrapperSmall: {
+        paddingLeft: Spacing.xSmall_8,
+        paddingRight: Spacing.xSmall_8,
+        borderRadius: Spacing.xxSmall_6,
+    },
+    wrapperLarge: {
+        paddingLeft: Spacing.small_12,
+        paddingRight: Spacing.small_12,
+        paddingTop: Spacing.xxSmall_6,
+        paddingBottom: Spacing.xxSmall_6,
+        borderRadius: Spacing.large_24,
+    },
+});
+
+const _generateColorStyles = (
```

Comment:
change: if you look at other instances where we are using this type of pattern, we initialize a styles object so that the styles are cached. without this, the styles are regenerated every time the component renders.

i think you forgot the `if else` block where we check if we have already created the style because i see it is initialized above.

if you're going to go with this approach, could you add in the caching?

Reply:
OMG that's a huge problem, great catch